src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -575,19 +576,24 @@ protected override TypeSymbol InferTypeOfVarVariable(DiagnosticBag diagnostics)
                 // in the factory method call for this symbol.
                 return ((ForEachLoopBinder)this.binder).InferCollectionElementType(diagnostics, _collection);
             }
+
+            internal override SyntaxNode ForbiddenZone => _collection;
         }
 
         /// <summary>
         /// Symbol for an out variable local that might require type inference during overload resolution.
         /// </summary>
         private class PossiblyImplicitlyTypedOutVarLocalSymbol : SourceLocalSymbol
         {
+            private readonly CSharpSyntaxNode _containingInvocation;
+
             public PossiblyImplicitlyTypedOutVarLocalSymbol(
                 Symbol containingSymbol,
                 Binder binder,
                 TypeSyntax typeSyntax,
                 SyntaxToken identifierToken,
-                LocalDeclarationKind declarationKind)
+                LocalDeclarationKind declarationKind,
+                CSharpSyntaxNode containingInvocation)
```

Comment:
For `PossiblyImplicitlyTypedDeconstructionLocalSymbol` below, the `ForbiddenZone` is computed by walking up the tree to find the right parent.
But for `PossiblyImplicitlyTypedOutVarLocalSymbol`, you rely on the creation to capture the node for invocation. Why not do the same and walk up the tree instead?